ddtrace/propagation/http.py

```diff
@@ -885,12 +890,70 @@ def _inject(span_context, headers):
         return headers
 
 
+class _BaggageHeader:
+    """Helper class to inject/extract Baggage Headers"""
+
+    safe_characters_key = "ABCDEFGHIJKLMNOPQRSTUVWXYZ" "abcdefghijklmnopqrstuvwxyz" "0123456789" "!#$%&'*+-.^_`|~"
+    safe_characters_value = (
+        "ABCDEFGHIJKLMNOPQRSTUVWXYZ" "abcdefghijklmnopqrstuvwxyz" "0123456789" "!#$%&'()*+-./:<>?@[]^_`{|}~"
+    )
+
+    @staticmethod
+    def _encode_key(key: str) -> str:
+        return urllib.parse.quote(str(key).strip(), safe=_BaggageHeader.safe_characters_key)
+
+    @staticmethod
+    def _encode_value(value: str) -> str:
+        return urllib.parse.quote(str(value).strip(), safe=_BaggageHeader.safe_characters_value)
+
+    @staticmethod
+    def _inject(span_context: Context, headers: Dict[str, str]) -> None:
+        baggage_items = span_context._baggage.items()
+        if not baggage_items:
+            return
+
+        if len(baggage_items) > DD_TRACE_BAGGAGE_MAX_ITEMS:
+            return
+
+        header_value = ",".join(
+            f"{_BaggageHeader._encode_key(key)}={_BaggageHeader._encode_value(value)}" for key, value in baggage_items
+        )
+
+        buf = bytes(header_value, "utf-8")
```

Comment:
This operation can raise an esception. We should wrap the encoding of the header in a try-except